site/content/contribute/webapp/unit-testing.md

```diff
@@ -0,0 +1,279 @@
+    ```javascript
+    test('should call updateSection on handleExpand', () => {
+        const newUpdateSection = jest.fn();
+        const wrapper = mountWithIntl(
+            <EmailNotificationSetting
+                {...baseProps}
+                updateSection={newUpdateSection}
+            />
+        );
+        wrapper.instance().handleExpand();
+
+        expect(newUpdateSection).toBeCalled();
+        expect(newUpdateSection).toHaveBeenCalledTimes(1);
+        expect(newUpdateSection).toBeCalledWith('email');
+    });
+    ```
+
+6. When a function is passed to a component via props, make sure to test if it gets called for a particular event call or its state changes.
+    ```javascript
+    test('should call functions on handleSubmit', () => {
+        const newOnSubmit = jest.fn();
+        const newUpdateSection = jest.fn();
+        const wrapper = mountWithIntl(
+            <EmailNotificationSetting
+                {...baseProps}
+                onSubmit={newOnSubmit}
+                updateSection={newUpdateSection}
+            />
+        );
+
+        wrapper.instance().handleSubmit();
+
+        expect(newOnSubmit).not.toBeCalled();
+        expect(newUpdateSection).toHaveBeenCalledTimes(1);
+        expect(newUpdateSection).toBeCalledWith('');
+
+        wrapper.find('#emailNotificationNever').simulate('change');
+        wrapper.instance().handleSubmit();
+
+        expect(newOnSubmit).toBeCalled();
+        expect(newOnSubmit).toHaveBeenCalledTimes(1);
+        expect(newOnSubmit).toBeCalledWith({enableEmail: 'false'});
+
+        expect(savePreference).toHaveBeenCalledTimes(1);
+        expect(savePreference).toBeCalledWith('notifications', 'email_interval', '0');
+    });
+    ```
+
+7. Test the component's internal or lifecycle methods by having different sets of props.
+    ```javascript
+    test('should pass componentWillReceiveProps', () => {
+        const nextProps = {
+            enableEmail: true,
+            emailInterval: 30
+        };
+        const wrapper = mountWithIntl(<EmailNotificationSetting {...baseProps}/>);
+        wrapper.setProps(nextProps);
+
+        expect(wrapper.state('enableEmail')).toBe(nextProps.enableEmail);
+        expect(wrapper.state('emailInterval')).toBe(nextProps.emailInterval);
+
+        ...
+        const shouldUpdate = wrapper.instance().shouldComponentUpdate({show: true});
+        expect(shouldUpdate).toBe(true);
+    });
+    ```
+
+8. Provide a mockup of a function required by the component but also pass other exported functions out of it to prevent potential error when those were used indirectly by another functions.
+    ```javascript
+    jest.mock('utils/utils', () => {
+        const original = require.requireActual('utils/utils');
+        return {
+            ...original,
+            isMobile: jest.fn(() => true),
+        };
+    });
+    ```
+
```

Comment:
Steps 8 and 9 look the same.

Reply:
Oh right, thanks @amyblais, I'll update that.

Reply:
Noe updated, please see screenshot, thanks
![screen shot 2018-11-28 at 9 59 00 pm](https://user-images.githubusercontent.com/5334504/49156777-2e58a380-f359-11e8-8aff-29ef101d2a9f.png)